lib/nerves_motd.ex

```diff
@@ -31,70 +32,126 @@ defmodule NervesMOTD do
   @spec print([option()]) :: :ok
   def print(opts \\ []) do
     {:ok, _} = Application.ensure_all_started(:nerves_runtime)
-    IO.puts(generate(opts))
+
+    IO.puts(logo(opts))
+
+    IO.puts(uname())
+
+    Enum.each(rows(), &IO.puts(["  ", format_row(&1)]))
+
+    IO.puts("""
+
+    Nerves CLI help: https://hexdocs.pm/nerves/using-the-cli.html
+    """)
+  end
+
+  @spec logo([option()]) :: iodata()
+  defp logo(opts) do
+    Keyword.get(opts, :logo, @logo)
   end
 
-  defp generate(opts) do
+  @spec rows() :: [[cell()]]
+  defp rows() do
     [
-      logo_text(opts),
-      uname(),
-      """
+      [{"Uptime", uptime()}],
+      [{"Clock", clock()}],
+      [],
+      [firmware_cell(), applications_cell()],
+      [memory_usage_cell(), {"Load average", load_average()}],
+      [{"Hostname", hostname()}, {"Networks", network_names()}]
+    ]
+  end
 
-        Uptime : #{uptime()}
-        Clock  : #{clock()}
+  @spec format_row([cell()]) :: iolist()
+  # A blank line
+  defp format_row([]), do: []
+
+  # A row with full width
+  defp format_row([{label, formatted_iodata}]) do
+    :io_lib.format("~-12ts : ~s", [label, formatted_iodata])
+  end
+
+  # A row with two columns
+  defp format_row([col0, col1]) do
+    [format_cell(col0, 0), format_cell(col1, 1)]
+  end
 
-        Firmware     : #{String.pad_trailing(firmware_text(), 24, " ")}\tApplications : #{application_text()}
-        Memory usage : #{String.pad_trailing(memory_usage_text(), 24, " ")}\tLoad average : #{load_average()}
-        Hostname     : #{String.pad_trailing(hostname_text(), 24, " ")}\tNetworks     : #{networks_text()}
+  @spec format_cell(cell(), 0 | 1) :: iolist()
+  defp format_cell({label, formatted_iodata}, column_index) do
+    case column_index do
+      0 -> "~-12ts : ~-24ts"
+      _ -> "~-12ts : ~s"
+    end
+    |> :io_lib.format([label, formatted_iodata])
+  end
 
-      Nerves CLI help: https://hexdocs.pm/nerves/using-the-cli.html
-      """
+  defp format_cell({label, formatted_iodata, :green}, column_index) do
+    [
+      :io_lib.format("~-12ts : ", [label]),
+      IO.ANSI.green(),
+      case column_index do
+        0 -> "~-24ts"
+        _ -> "~s"
+      end
+      |> :io_lib.format([formatted_iodata]),
+      IO.ANSI.reset()
     ]
   end
 
-  defp logo_text(opts) do
-    Keyword.get(opts, :logo, @logo)
+  defp format_cell({label, formatted_iodata, :red}, column_index) do
+    [
+      :io_lib.format("~-12ts : ", [label]),
+      IO.ANSI.red(),
+      case column_index do
+        0 -> "~-24ts"
+        _ -> "~s"
+      end
+      |> :io_lib.format([formatted_iodata]),
+      IO.ANSI.reset()
+    ]
   end
 
```

Comment:
```suggestion
  defp ansi(:red), do: IO.ANSI.red()
  defp ansi(:green), do: IO.ANSI.green()
```
I'm not sure if I commented this right, but my intent was to combine the `:red` and `:green` functions and then have an `ansi/1` function that inserts the right escape sequence.

Reply:
Ah I know what you mean:bulb: The two functions are almost the same. Also we could support different supported colors when needed in the future. I was able to do the refactoring using `apply/3` like `apply(IO.ANSI, color, [])`. How does it look to you?

Reply:
Yes, I like that.